### PR TITLE
[WCM][Validator] Update datetime validator

### PR DESCRIPTION
**Pull request by lepiaf**
Documentation for https://github.com/symfony/symfony/issues/11925

### CODE REVIEW

File: reference/constraints/Timestamp.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
+Timestamp
+=========
+
+Validates that a value is a valid that follows a specific format.
```
Comment (HeahDude) [inline reference/constraints/Timestamp.rst]: I would suggest something like: `Validates a timestamp for a specific format.`

File: reference/constraints/Timestamp.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+This option allows to validate a custom date format. See
+:phpmethod:`DateTime::createFromFormat` for formatting options.
+
+message
+~~~~~~~
+
+**type**: ``string`` **default**: ``This value is not a valid timestamp.``
+
+This message is shown if the underlying data is not a valid date.
```
Comment (HeahDude) [inline reference/constraints/Timestamp.rst]: `not in a valid state`.

**Review by HeahDude — COMMENTED** [review]
Thanks for fixing that old issue :)

**Review by javiereguiluz — COMMENTED** [review]
I'm sorry but I find this doc and validator very confusing.

Why?

* People think of timestamps as a UNIX timestamp (e.g. `1256953732`)
* This is not correct because a "timestamp" is just any date + time (see https://en.wikipedia.org/wiki/Timestamp)
* In the doc, it says that the "format" options is for "validating a custom date format" and "message" is used when "is not a valid date"

So, we're mixing "date" with "timestamp". By the way, why do we need this validator? Is not enough to use "DateTime", given that a timestamp is just a "date + time" ?

**Comment by lepiaf** [comment]
We want to regroup all specific validator on Date (TimeValidator, DateValidator and DateTimeValidator) into one and unique validator: TimestampValidator.

These 3 validators do almost the same jobs: check that value is compliant according to format. 

With Timestamp validator, you could check that value is a real timestamp `Assert\Timestamp("U")` without creating a custom validator.

Maybe I did not explain correctly in document, but for reference: https://github.com/symfony/symfony/issues/11925#issue-42782573

**Comment by javiereguiluz** [comment]
@lepiaf I see. Thanks for the explanation! It's clear to me now ... but I expect a lot of confusion about this naming, so I'm going to add a comment in the related PR to ask for feedback.

**Review by roman-1983 — APPROVED** [review]

**Pull request closed without merging**